gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FlowSpec.java

```diff
@@ -518,15 +530,5 @@ public static int maxFlowSpecUriLength() {
       return URI_SCHEME.length() + ":".length() // URI separator
         + URI_PATH_SEPARATOR.length() + ServiceConfigKeys.MAX_FLOW_NAME_LENGTH + URI_PATH_SEPARATOR.length() + ServiceConfigKeys.MAX_FLOW_GROUP_LENGTH;
     }
-
-    /**
-     * Create a new FlowSpec object with the added property defined by path and value parameters
-     * @param path key for new property
-     * @param value
-     */
-    public static FlowSpec createFlowSpecWithProperty(FlowSpec flowSpec, String path, String value) {
-      Config updatedConfig = flowSpec.getConfig().withValue(path, ConfigValueFactory.fromAnyRef(value));
-      return new Builder(flowSpec.getUri()).withConfig(updatedConfig).build();
```

Comment:
my hunch is that this `.build()` is what generated all the garbage.  most likely the conversion of config to props